File: 2 Pointers/Valid Palindrome/Solution.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int leftPos = 0;
        int rightPos = s.size() - 1;

        while (leftPos < rightPos) {
            
            if (!isalnum(s[leftPos])) {
                leftPos++;
                continue;
            } 
            if (!isalnum(s[rightPos])) {
                rightPos--;
                continue;
            }

            if (tolower(s[leftPos]) != tolower(s[rightPos])) {
                return false;
            }
            
            leftPos++;
            rightPos--;
        }
        return true;
    }
};

File: 2 Pointers/Valid Palindrome/Solution.kt
class Solution {
    fun isPalindrome(s: String): Boolean {
        var leftPos = 0
        var rightPos = s.length - 1

        while (leftPos < rightPos) {
            if (!s[leftPos].isLetterOrDigit()) {
                leftPos++
                continue
            }
            if (!s[rightPos].isLetterOrDigit()) {
                rightPos--
                continue
            }
            if (s[leftPos].lowercaseChar() != s[rightPos].lowercaseChar()) {
                return false
            }
            leftPos++
            rightPos--
        }
        return true
    }
}